rbac/management/utils.py

```diff
@@ -84,11 +84,11 @@ def get_principal(username, request, verify_principal=True, from_query=False):
             principal = Principal.objects.get_or_create(
                 username=user.username, tenant=tenant, type=SERVICE_ACCOUNT_KEY, service_account_id=user.client_id
             )
-
-        # Avoid possible race condition if the user was created while checking BOP
-        principal, created = Principal.objects.get_or_create(
-            username=username, tenant=tenant
-        )  # pylint: disable=unused-variable
+        else:
```

Comment:
tests are failing because this piece of code should be outside the else block + tuple of 2 values is returned from `Principal.objects.get_or_create()`, the principal object and then boolean (`True` if object was created, `False` if object was already in db) so we need 2 variables on the left side, in the python you can use `_` as variable name and then this variable is skipped by linters

```
        user: User = request.user
        if user and user.is_service_account:
            Principal.objects.get_or_create(
                username=user.username, tenant=tenant, type=SERVICE_ACCOUNT_KEY, service_account_id=user.client_id
            )
        # Avoid possible race condition if the user was created while checking BOP
        principal, _ = Principal.objects.get_or_create(username=username, tenant=tenant)
```

Reply:
Thanks for the review!

> tests are failing because this piece of code should be outside the else block

Actually I think the code is properly indented, but I wanted to avoid running two very similar queries for the same outcome and the same purpose. I would vote for keeping it indented inside the `else:` block, what do you think?

> tuple of 2 values is returned from Principal.objects.get_or_create(), the principal object and then boolean (True if object was created, False if object was already in db) so we need 2 variables on the left side, in the python you can use _ as variable name and then this variable is skipped by linters

You are right, thanks so much! I've added those variable placeholders and now tests pass! :smile:

Reply:
yes you are right with the "else" block 👍 now I can see it